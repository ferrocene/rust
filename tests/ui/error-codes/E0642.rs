//@ revisions: edition2015 edition2018
//@ [edition2015] edition: 2015
//@ [edition2018] edition: 2018..
//@ [edition2015]run-rustfix

#![allow(unused)] // for rustfix

#[derive(Clone, Copy)]
struct S;

trait T {
    fn foo((x, y): (i32, i32)); //[edition2015]~ ERROR patterns aren't allowed in methods without bodies
                                //[edition2018]~^ ERROR patterns aren't allowed in functions without bodies

    fn bar((x, y): (i32, i32)) {} //[edition2015]~ ERROR patterns aren't allowed in methods without bodies

    fn method(S { .. }: S) {} //[edition2015]~ ERROR patterns aren't allowed in methods without bodies

    fn f(&ident: &S) {} // ok
    fn g(&&ident: &&S) {} // ok
    fn h(mut ident: S) {} // ok
}

fn main() {}
