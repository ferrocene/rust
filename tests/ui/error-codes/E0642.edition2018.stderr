error[E0642]: patterns aren't allowed in functions without bodies
  --> $DIR/E0642.rs:12:12
   |
LL |     fn foo((x, y): (i32, i32));
   |            ^^^^^^ pattern not allowed in function without body

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0642`.
